exec/java-exec/src/main/java/org/apache/drill/metastore/TableMetadata.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.metastore;
+
+import org.apache.drill.common.expression.SchemaPath;
+import org.apache.drill.exec.record.metadata.ColumnMetadata;
+import org.apache.drill.exec.record.metadata.SchemaPathUtils;
+import org.apache.drill.exec.record.metadata.TupleSchema;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Metadata which corresponds to the table level.
+ */
+public class TableMetadata implements BaseMetadata {
+  private final String tableName;
+  private final String location;
+  private final TupleSchema schema;
+  private final Map<SchemaPath, ColumnStatistics> columnsStatistics;
+  private final Map<String, Object> tableStatistics;
+  private final long lastModifiedTime;
+  private final String owner;
+  private final Set<String> partitionKeys;
```

Comment:
Conceptually, the fact that partitionKeys belongs here makes me think what about other physical attributes of a table ?     Such as primary keys,  indexing columns etc.  However, a distinction is that here TableMetadata is only targeting file system tables, not tables that are native to other storage/format plugins.  For example, MapR-DB tables will have their own version of TableMetadata which may not match with this class definition.   Are you thinking that TableMetadata will be the universal table metadata for **ALL** types of tables regardless of their data source ?

Reply:
Agree, will create the interface with name `TableMetadata`, move base methods there and rename this class into `FileTableMetadata`.